### PR TITLE
fix menu items= API

### LINKED ISSUE #1903: New rx.menu() API does not work correctly
**Describe the bug**
When using `rx.menu()`, I have to provide at least the arguments `button` and `items`, which are both type-hinted as `Unknown`. When I provide an avatar as the button and a list of strings as the items, the resulting menu only contains one item (named after all the strings combined) which is also unusable. 

**To Reproduce**
 ```python
def avatar_menu():
    return rx.menu(button=rx.avatar(name="Max Hustemann"), items=["A", "B", "C"])
```

**Expected behavior**
A menu that contains the selectable items A, B and C.

**Specifics (please complete the following information):**
 - Python Version: 3.11.5
 - Reflex Version: 0.2.8
 - OS: Fedora 
 - Browser (Optional): Brave
 
**Additional context**
I suspect this happens because no `MenuItem` is ever created for each item and a `MenuList` is instead used directly. Also, I'd suggest to type hint the items as a `List`.

### PR DESCRIPTION
**Pull request by Lendemor**
Fix #1903

### CODE REVIEW

File: reflex/components/overlay/menu.py [hunk 1]
```diff
@@ -74,7 +74,13 @@ def get_event_triggers(self) -> dict[str, Union[Var, Any]]:
         }
 
     @classmethod
-    def create(cls, *children, button=None, items=None, **props) -> Component:
+    def create(
+        cls,
+        *children,
+        button: Optional[Component] = None,
+        items: Optional[List] = None,
```
Comment (masenf) [inline reflex/components/overlay/menu.py:81]: is it possible for `items` to be a `Var`? doesn't look like that case is handled

Comment (Lendemor) [reply reflex/components/overlay/menu.py:81]: It wasn't supported initially, should I add this in this PR, or I can add it in a separate one.

Comment (masenf) [reply reflex/components/overlay/menu.py:81]: i'm fine either way, i just anticipate that coming as the next question 😂

Comment (picklelo) [reply reflex/components/overlay/menu.py:81]: We can make a new PR for this

**Review by picklelo — APPROVED** [review]

**Pull request merged**